### DISCUSSION TITLE
Parameter Apa Yang dapat Mentuning Model

### DISCUSSION THREAD

**Discussion by rulierinton** [teamalgoritma/community#160] [Q&A] [answered]
Hai Bapak & Ibu,

Saya sendang mengerjakan Capstone tapi untuk peformanya sulit sekali mendapatkan hasil yang sudah ditentukan, saya sudah mencoba mengubah paramaeter pada model sperti dibawah ini:

ctrl <- trainControl(method = "repeatedcv",
                     number = 5, # Saya ubah 1 - 15
                      repeats = 15) # Saya ubah 1 - 15
 model_rf <- train(target ~ .,
                     data = data,
                    method = "rf", 
                   trControl = ctrl)

setelah saya ubah parameter tersebut yang membedakannya hanya waktu proses saja yang makin lama, untuk hasilnya terkadang semakin baik atau malah terjadi penurunan dibandingkan parameter yang lebigh kecil 

terimakasih,
Ruli

**Comment by rulierinton** [comment]
ctrl <- trainControl(method = "repeatedcv",
                     number = 6, 
                      repeats = 80) 
 
model_rf <- train(strength ~.,
                     data = data_train ,
                    method = "rf",
                   trControl = ctrl)

saya sudah menggunakan repeats 80 tapi hanya niali MAE saja yang terimprove tapi nilai R-Square tidak ada perubahan sama sekali

**Comment by NabiilahArdini** [comment, marked as answer]
Halo @rulierinton ,

Penambahan repeats / fold memang tidak selalu meningkatkan performa model. Bila ketika ditingkatkan repeats/foldnya performanya tidak meningkat, maka model sudah convergen / stabil / mencapai titik jenuh / terbaiknya untuk data training yang diberikan. Bahkan ketika performanya sudah menurun, maka ada indikasi terjadi overfitting. 

Untuk itu tuning yang diperlukan mungkin lebih berhubungan dengan data, seperti pembuatan prediktor tambahan/scaling data.

Semoga insight ini bermanfaat!
